File: articles/future-knowledge-platforms/fortran/future_knowledge_platform_metrics.f90
program future_knowledge_platform_metrics
  implicit none

  real :: objects
  real :: relationships

  objects = 14.0
  relationships = 14.0

  print *, "Future Knowledge Platform Metrics"
  print *, "Metadata coverage: ", 13.0 / objects
  print *, "Provenance coverage: ", 12.0 / objects
  print *, "Reuse context coverage: ", 11.0 / objects
  print *, "Review context coverage: ", 11.0 / objects
  print *, "Relationship traceability: ", 13.0 / relationships
  print *, "Underspecified relationship risk: ", 1.0 / relationships
end program future_knowledge_platform_metrics

File: articles/future-knowledge-platforms/cpp/future_knowledge_platform_metrics.cpp
#include <iostream>
#include <iomanip>

int main() {
    double objects = 14.0;
    double relationships = 14.0;

    std::cout << std::fixed << std::setprecision(3);
    std::cout << "Metadata coverage: " << 13.0 / objects << "\n";
    std::cout << "Provenance coverage: " << 12.0 / objects << "\n";
    std::cout << "Reuse context coverage: " << 11.0 / objects << "\n";
    std::cout << "Review context coverage: " << 11.0 / objects << "\n";
    std::cout << "Relationship traceability: " << 13.0 / relationships << "\n";
    std::cout << "Underspecified relationship risk: " << 1.0 / relationships << "\n";

    return 0;
}
